lib/websocket.js

```diff
@@ -1,22 +1,69 @@
 'use strict';
 
 var levels = require('./levels');
-
+var times = require('./times');
+var calcData = require('./data/calcdelta');
+var ObjectID = require('mongodb').ObjectID;
+var profilefunctions = require('./profilefunctions')();
+       
 function init (env, ctx, server) {
 
   function websocket ( ) {
     return websocket;
   }
 
+  //var log_yellow = '\x1B[33m';
+  var log_green = '\x1B[32m';
+  var log_magenta = '\x1B[35m';
+  var log_reset = '\x1B[0m';
+  var LOG_WS = log_green + 'WS: ' + log_reset;
+  var LOG_DEDUP = log_magenta + 'DEDUPE: ' + log_reset;
+  
   var io;
   var watchers = 0;
   var lastData = {};
+  var lastProfileSwitch = null;
 
-  var alarmType2Level = {
-    urgent_alarm: levels.URGENT
-    , alarm: levels.WARN
+  // TODO: this would be better to have somehow integrated/improved
+  var supportedCollections = { 
+  'treatments' : env.treatments_collection, 
+  'entries': env.mongo_collection, 
+  'devicestatus': env.devicestatus_collection, 
+  'profile': env.profile_collection, 
+  'food': env.food_collection
   };
 
+  // This is little ugly copy but I was unable to pass testa after making module from status and share with /api/v1/status
+  function status (profile) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["profile" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2651207102/issues/source?bid=2452374&fileBranchId=3324571#l37)